### ISSUE TITLE
Inconsistencies with CMIP6

### ISSUE THREAD

**Issue by sol1105** [WCRP-CORDEX/cordex-cmip6-cmor-tables#66]
Hi @larsbuntemeyer 

I just want to list a few inconsistencies I noticed between CORDEX-CMIP6 and CMIP6 CMOR tables. Some are or may be relevant for a successful application of the CMOR library:

- The `cell_methods` `area: mean time:mean` in CORDEX-CMIP6 are specified as `area: time: mean` in CMIP6 CMOR tables.
- The positive attribute for fluxes like `hfls, hfss, rlds, rlus`, etc. is not defined in the CORDEX-CMIP6 CMOR tables. Even though the `standard_name` includes the correct orientation, the attribute is important for the CMOR application. CMOR does not write the `positive` attribute to the file, but uses it to check and potentially switch the orientation.
- In the recent PR #63, you updated the order of `dimensions` in the CMOR tables. However, in CMIP6, singleton vertical dimensions (single vertical levels such as `p220` or `height2m`) are always listed last, eg. `longitude latitude time height2m`. Only "normal" vertical dimensions (like `alevel`, `plev19` etc.) are listed before the `time` dimension. I am not sure if this has relevance for the CMOR application.
- The `comment` attribute is mostly empty, one could use the comments defined in CMIP6.
- The `modeling_realm` is set to `atmos` in all cases, even for variables such as `mrso` with comment `The mass of water in all phases per unit area, summed over all soil layers.`. Beside the `modeling_realm`, also the  `cell_methods` for this kind of variables are different in CMIP6: `"area: mean where land time: mean"` instead of `area: mean time: mean`.

Lastly, I noticed a small typo for `od550aer`:
both the `label` and the `out_name` for od550aer are followed by a space:       
``` 
"od550aer_": {
            [...]
            "out_name": "od550aer_",
            [...]
```

**Comment by larsbuntemeyer** [comment]
Thanks again @sol1105 very much appreciated! I'll split up your issues and work on that on our [data request table](https://github.com/WCRP-CORDEX/data-request-table). Just some few comments here:

* I really wasn't aware of the difference between "normal" vertical dimensions and singleton vertical dimensions, thanks for that! @gnikulin Could you confirm that the soil variables in CORDEX-CMIP6 (`tsl`, `mrsfl` and `mrsol`) are supposed to be "real" 3D variables (as it is mentioned in the comments [here](https://docs.google.com/spreadsheets/d/1qUauozwXkq7r1g-L4ALMIkCNINIhhCPx/#gid=1672965248)). They weren't part of the CMOR in tables in CMIP5 so i am not sure. 

* We discussed the `modeling_realm` issue in https://github.com/WCRP-CORDEX/data-request-table/issues/4. @gnikulin I think our last decision on that was to simply remove it from the variable attributes?

I will add a test case with a vertical coordinate to the [examples](https://wcrp-cordex.github.io/cordex-cmip6-cmor-tables/cmor-examples.html#cmorization-example) to make sure the CMOR tool works with it. The examples are created using the python API without the `_cmip6_option` (https://github.com/PCMDI/cmor/discussions/679#discussioncomment-3842956)) in the input dataset table.

**Comment by larsbuntemeyer** [comment]
@gnikulin should we include the original CMIP6 variable attribute "comment"? I would vote for that. Right now, we use the "Comments" column from the [original CMIP6 data request](https://docs.google.com/spreadsheets/d/1qUauozwXkq7r1g-L4ALMIkCNINIhhCPx).

**Comment by larsbuntemeyer** [comment]
By comparing cell_methods of current CORDEX data request with CMIP6 data request, i find the following inconsistencies:

| out_name   | frequency   | cordex                 | cmip6                             | cmip6_realm   |
|:-----------|:------------|:-----------------------|:----------------------------------|:--------------|
| evspsbl    | day         | area: time: mean       | area: mean where land time: mean  | land          |
| tsl        | mon         | area: time: mean       | area: mean where land time: mean  | land          |
| tsl        | day         | area: time: mean       | area: mean where land time: mean  | land          |
| tsl        | 6hrPt       | area: mean time: point | area: mean where land time: point | land          |
| mrros      | mon         | area: time: mean       | area: mean where land time: mean  | land          |
| mrros      | day         | area: time: mean       | area: mean where land time: mean  | land          |
| mrro       | mon         | area: time: mean       | area: mean where land time: mean  | land          |
| mrro       | day         | area: time: mean       | area: mean where land time: mean  | land          |
| snm        | day         | area: time: mean       | area: mean where land time: mean  | landIce land  |
| mrfso      | mon         | area: time: mean       | area: mean where land time: mean  | land landIce  |
| mrsfl      | mon         | area: time: mean       | area: mean where land time: mean  | land          |
| mrsfl      | day         | area: time: mean       | area: mean where land time: mean  | land          |
| mrso       | mon         | area: time: mean       | area: mean where land time: mean  | land          |
| mrso       | day         | area: time: mean       | area: mean where land time: mean  | land          |
| mrsos      | mon         | area: time: mean       | area: mean where land time: mean  | land          |
| mrsos      | day         | area: time: mean       | area: mean where land time: mean  | land          |
| mrsol      | mon         | area: time: mean       | area: mean where land time: mean  | land          |
| mrsol      | day         | area: time: mean       | area: mean where land time: mean  | land          |
| snw        | mon         | area: time: mean       | area: mean where land time: mean  | landIce land  |
| snw        | day         | area: time: mean       | area: mean where land time: mean  | landIce land  |
| snw        | 6hrPt       | area: mean time: point | area: mean where land time: point | landIce land  |
| snd        | mon         | area: time: mean       | area: mean where land time: mean  | landIce land  |
| snd        | day         | area: time: mean       | area: mean where land time: mean  | landIce land  |
| evspsblpot | mon         | area: time: mean       | area: mean where land time: mean  | land          |
| evspsblpot | day         | area: time: mean       | area: mean where land time: mean  | land          |
| mrsofc     | fx          | area: mean             | area: mean where land             | land          |

@gnikulin Should we stick with CMIP6 standard here? Note, that we don't have a modeling_realm in CORDEX for now...

**Comment by gnikulin** [comment]
>     * I really wasn't aware of the difference between "normal" vertical dimensions and singleton vertical dimensions, thanks for  that! @gnikulin Could you confirm that the soil variables in CORDEX-CMIP6 (`tsl`, `mrsfl` and `mrsol`) are supposed to be  "real" 3D variables (as it is mentioned in the comments [here](https://docs.google.com/spreadsheets/d/1qUauozwXkq7r1g-L4ALMIkCNINIhhCPx/#gid=1672965248)). They weren't part of the CMOR in tables in CMIP5 so i am not sure.

Yes, all 3 variables (`tsl`, `mrsfl` and `mrsol`)  are "real" 3D (each soil layer).

**Comment by gnikulin** [comment]
> @gnikulin should we include the original CMIP6 variable attribute "comment"? I would vote for that (although that might depend on the table_id if i remember correctly). Right now, we use the "Comments" column from the [original CMIP6 data request](https://docs.google.com/spreadsheets/d/1qUauozwXkq7r1g-L4ALMIkCNINIhhCPx).

Yes, I would also use "Comments" from CMIP6 when possible. We need to make minor adjustments because there are several inconsistencies in the CMIP6 comments attribute for some "similar" variables.

**Comment by gnikulin** [comment]
> By comparing cell_methods of current CORDEX data request with CMIP6 data request, i find the following inconsistencies (more details here: [WCRP-CORDEX/data-request-table#19](https://github.com/WCRP-CORDEX/data-request-table/issues/19))
> out_name 	frequency 	cordex 	cmip6 	cmip6_realm 	cmip6_table
> evspsbl 	day 	area: time: mean 	area: mean where land time: mean 	land 	Eday
> tsl 	mon 	area: time: mean 	area: mean where land time: mean 	land 	Lmon
> tsl 	day 	area: time: mean 	area: mean where land time: mean 	land 	Eday
> tsl 	6hrPt 	area: mean time: point 	area: mean where land time: point 	land 	6hrPlevPt
> mrros 	mon 	area: time: mean 	area: mean where land time: mean 	land 	Lmon
> mrros 	day 	area: time: mean 	area: mean where land time: mean 	land 	Eday
> mrro 	mon 	area: time: mean 	area: mean where land time: mean 	land 	Lmon
> mrro 	day 	area: time: mean 	area: mean where land time: mean 	land 	day
> snm 	day 	area: time: mean 	area: mean where land time: mean 	landIce land 	Eday
> mrfso 	mon 	area: time: mean 	area: mean where land time: mean 	land landIce 	Lmon
> mrsfl 	mon 	area: time: mean 	area: mean where land time: mean 	land 	Emon
> mrsfl 	day 	area: time: mean 	area: mean where land time: mean 	land 	Eday
> mrso 	mon 	area: time: mean 	area: mean where land time: mean 	land 	Lmon
> mrso 	day 	area: time: mean 	area: mean where land time: mean 	land 	day
> mrsos 	mon 	area: time: mean 	area: mean where land time: mean 	land 	Lmon
> mrsos 	day 	area: time: mean 	area: mean where land time: mean 	land 	day
> mrsol 	mon 	area: time: mean 	area: mean where land time: mean 	land 	Emon
> mrsol 	day 	area: time: mean 	area: mean where land time: mean 	land 	Eday
> snw 	mon 	area: time: mean 	area: mean where land time: mean 	landIce land 	LImon
> snw 	day 	area: time: mean 	area: mean where land time: mean 	landIce land 	day
> snw 	6hrPt 	area: mean time: point 	area: mean where land time: point 	landIce land 	6hrPlevPt
> snd 	mon 	area: time: mean 	area: mean where land time: mean 	landIce land 	LImon
> snd 	day 	area: time: mean 	area: mean where land time: mean 	landIce land 	Eday
> evspsblpot 	mon 	area: time: mean 	area: mean where land time: mean 	land 	Emon
> evspsblpot 	day 	area: time: mean 	area: mean where land time: mean 	land 	Eday
> mrsofc 	fx 	area: mean 	area: mean where land 	land 	fx
> 
> @gnikulin Should we stick with CMIP6 standard here? Note, that we don't have a modeling_realm in CORDEX for now...

We need to check in detail. For all "soil" variables  we indeed should use `area: mean where land`. However, for example `evspsbl` in the CMIP5 Amon table has   "cell_methods": "area: time: mean" and is provided over both land and water.
